setup.py

```diff
@@ -297,67 +297,70 @@ def build_extension(self, ext):
-        if (
-            sys.version_info >= (3, 6, 0)
-            and ext.name == "ddtrace.appsec.iast._taint_tracking._native"
-            and os.path.exists(cmake_list_path)
-        ):
-            import shutil
-
-            os.makedirs(tmp_iast_path, exist_ok=True)
-
-            import subprocess
-
-            cmake_command = os.environ.get("CMAKE_COMMAND", "cmake")
-            build_type = "RelWithDebInfo" if DEBUG_COMPILE else "Release"
-            build_args = ["--config", build_type]
-            cmake_args = [
-                "-S",
-                IAST_DIR,
-                "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY={}".format(tmp_iast_path),
-                "-B",
-                tmp_iast_path,
-                "-DPYTHON_EXECUTABLE={}".format(sys.executable),
-                "-DCMAKE_BUILD_TYPE={}".format(build_type),
-            ]
-
-            if CURRENT_OS == "Windows":
-                cmake_args.extend(["-A", "x64" if platform.architecture()[0] == "64bit" else "Win32"])
-
-            if CURRENT_OS == "Darwin" and sys.version_info >= (3, 8, 0):
-                # Cross-compile support for macOS - respect ARCHFLAGS if set
-                # Darwin Universal2 should bundle both architectures
-                archs = re.findall(r"-arch (\S+)", os.environ.get("ARCHFLAGS", ""))
-                if archs:
-                    cmake_args += [
-                        "-DBUILD_MACOS=ON",
-                        "-DCMAKE_OSX_ARCHITECTURES={}".format(";".join(archs)),
-                    ]
-
-            # Set CMAKE_BUILD_PARALLEL_LEVEL to control the parallel build level
-            # across all generators.
-            if "CMAKE_BUILD_PARALLEL_LEVEL" not in os.environ:
-                # self.parallel is a Python 3 only way to set parallel jobs by hand
-                # using -j in the build_ext call, not supported by pip or PyPA-build.
-                if hasattr(self, "parallel") and self.parallel:
-                    # CMake 3.12+ only.
-                    build_args += ["-j{}".format(self.parallel)]
-
-            cmake_cmd_with_args = [cmake_command] + cmake_args
-            subprocess.run(cmake_cmd_with_args, cwd=tmp_iast_path, check=True)
-
-            build_command = [cmake_command, "--build", tmp_iast_path] + build_args
-            subprocess.run(build_command, cwd=tmp_iast_path, check=True)
-
-            for directory_to_remove in ["_deps", "CMakeFiles"]:
-                shutil.rmtree(os.path.join(tmp_iast_path, directory_to_remove))
-            for file_to_remove in ["Makefile", "cmake_install.cmake", "compile_commands.json", "CMakeCache.txt"]:
-                if os.path.exists(os.path.join(tmp_iast_path, file_to_remove)):
-                    os.remove(os.path.join(tmp_iast_path, file_to_remove))
-
-            shutil.copy(os.path.join(IAST_DIR, tmp_filename), tmp_iast_file_path)
-        else:
-            build_ext.build_extension(self, ext)
+        try:
```

Comment:
A large `try` block like this one makes it hard to tell by reading which part of the block is expected to raise an exception. You might consider putting the `try` around only the specific bits of logic that you expect to raise exceptions.

Reply:
you're right, good comment! :) I'll update this try-except

Reply:
ready @emmettbutler :+1: